pilot/pkg/networking/util/util.go

```diff
@@ -708,3 +713,19 @@ func TraceOperation(host string, port int) string {
 	// Format : "%s:%d/*"
 	return DomainName(host, port) + "/*"
 }
+
+// CheckProxyVerionForMX checks whether metadata exchange filters should be injected
+// based on proxy version and presence of the well known metadata exchange EnvoyFilter.
+// If those EnvoyFilter presents (which is the case if user in place upgrade with istioctl )
+// TODO: Remove this at 1.14 release, since we support skip version upgrade
+//       now and 1.11 proxy can talk to 1.13 control plane.
+func CheckProxyVerionForMX(push *model.PushContext, proxyVersion *model.IstioVersion) bool {
+	hasDefaultEnvoyFilter := push.HasEnvoyFilters(MXName111, push.Mesh.RootNamespace) || push.HasEnvoyFilters(MXName110, push.Mesh.RootNamespace)
```

Comment:
I realize this is a huge edge case to have mx10 but not mx11 or vise versa, but wouldn't ideal behavior be:

```
if ver >= 1.12 { return true }
if ver >= 1.11 { return !has(mx11) }
if ver >= 1.10 { return !has(mx10) }
...
```

(where true = use native code)

Reply:
Also want to note there could be `metadata-exchange-1.10-<revision>`, hypothetically. This should be impossible unless someone in place upgraded a revision or something which is not supported. I don't think we need to handle the case, just mentioning it.

Reply:
Sure, I've updated the code as you suggested, and yeah I just ignored revision assuming user will always use revision.